README.md

```diff
@@ -23,84 +23,149 @@ Dependencies
+                self.outputs.append(
+                    [data['channel'], 'from repeat1 "{}" in channel {}'.format(
+                        data['text'], data['channel']
+                    )]
+                )
+
+The repeat plugin will now be loaded by the bot on startup. Run `rtmbot` from console to start your RtmBot.
+
+    rtmbot
 
 Create Plugins
 --------
 
 ####Incoming data
-Plugins are callback based and respond to any event sent via the rtm websocket. To act on an event, create a function definition called process_(api_method) that accepts a single arg. For example, to handle incoming messages:
+All events from the RTM websocket are sent to the registered plugins. To act on an event, create a function definition, inside your Plugin class, called process_(api_method) that accepts a single arg for data. For example, to handle incoming messages:
 
-    def process_message(data):
+    def process_message(self, data):
         print data
 
 This will print the incoming message json (dict) to the screen where the bot is running.
 
-Plugins having a method defined as ```catch_all(data)``` will receive ALL events from the websocket. This is useful for learning the names of events and debugging.
+Plugins having a method defined as ```catch_all(self, data)``` will receive ALL events from the websocket. This is useful for learning the names of events and debugging.
+
+For a list of all possible API Methods, look here: https://api.slack.com/rtm
 
 Note: If you're using Python 2.x, the incoming data should be a unicode string, be careful you don't coerce it into a normal str object as it will cause errors on output. You can add `from __future__ import unicode_literals` to your plugin file to avoid this.
 
 ####Outgoing data
-Plugins can send messages back to any channel, including direct messages. This is done by appending a two item array to the outputs global array. The first item in the array is the channel ID and the second is the message text. Example that writes "hello world" when the plugin is started:
 
-    outputs = []
-    outputs.append(["C12345667", "hello world"])
+#####RTM Output
+Plugins can send messages back to any channel or direct message. This is done by appending a two item array to the Plugin's output array (```myPluginInstance.output```). The first item in the array is the channel or DM ID and the second is the message text. Example that writes "hello world" when the plugin is started:
+
+    class myPlugin(Plugin):
+
+        def process_message(self, data):
+            self.outputs.append(["C12345667", "hello world"])
+
+#####SlackClient Web API Output
+Plugins also have access to the connected SlackClient instance for more complex output (or to fetch data you may need).
+
+    def process_message(self, data):
+        self.slack_client.api_call(
+            "chat.postMessage", channel="#general", text="Hello from Python! :tada:",
+            username='pybot', icon_emoji=':robot_face:'
```

Comment:
`username='pybot', icon_emoji=':robot_face:'`
to ` username="pybot", icon_emoji=":robot_face:"`
for consistency